### ISSUE TITLE
Windows 7 build and GitHub desktop

### ISSUE THREAD

**Issue by stevegdavis** [mapsme/omim#474]
I understand that this is not a direct app issue but I wondered if anyone on here has tried to do the same. I am trying to do a build on Windows 7:-
I have installed the GitHub desktop s/w OK and I have then clicked on the Clone in Desktop button and (after a few attempts) it downloaded the master branch successfully.
I have installed Android Studio, SDK, NDK, JDK etc.. (but not QT yet).
However when I open Android Studio for the omim-master project it opens numerous instances of the GitHub desktop application, not sure if this relates to the number of branches - just wondered if anyone else has seen this. 
Thanks.
![omim-github](https://cloud.githubusercontent.com/assets/15140454/11031177/7ffe4dc4-86ca-11e5-994e-4c670760e148.JPG)

**Comment by deathbaba** [comment]
Our team uses Mac OS for development, so we can’t suggest anything now.
Btw, you need to install Qt and have a path to qmake in PATH environment variable. And everything what is done in configure.sh script should be done too on Windows (see #468 for more details).

**Comment by stevegdavis** [comment]
OK thanks for the reply, I have just download the master branch as a zip file so I am not using GitHub desktop at all. QT installed and I have created the private.h & secure.properties file by hand.
However I am having build problems as in link #468, I have got over the "..\omim-master\android\res\values-zh-rHK\strings.xml:1:1: Error: Content is not allowed in prolog" problem (that #468 is having) by modifying the replace_links.bat file to run in windows, but now I am getting the same error as #468:-
:cppLibsBuildRelease FAILED          
              
FAILURE: Build failed with an exception.
              
* What went wrong:
Execution failed for task ':cppLibsBuildRelease'.
> A problem occurred starting process 'command 'bash''
but I have qmake in the widows path so maybe I should install Cygwin ?

**Comment by trashkalmar** [comment]
@stevegdavis Yes, you must have Cygwin with developer tools installed.
BTW, what did you modify in replace_links.bat? This file is intended to be run under Windows already.

**Comment by trashkalmar** [comment]
Also qmake should be located in the path listed in tools\autobuild\detect_qmake.sh

**Comment by stevegdavis** [comment]
I modified the replace_links.bat as running it in a command window it did not understand commands such as cp so replaced with copy and replaced  / path separator(s) with \ as appropriate. Also I have already included qmake in the detect_qmake.sh file (not sure if I have the syntax correct) but I do not get any direct errors in the AS build saying qmake not found so I assume the build system can see it ?Anyway I assume that detect_qmake.sh will have to be run in the Cynwin console ? as if I try to run it in Android Studio (right click on the file and with plugin installed) it just says "Executable not specified".
Thanks for your reply.

**Comment by trashkalmar** [comment]
detect_qmake.sh is part of build system and should not be run separately.
Also you have to install Visual Studio 2013 or newer (trial is ok) because it provides linker tool for building core.

**Comment by stevegdavis** [comment]
OK thanks for that, I have VS2010 pro installed so I assume that is not enough ? so if I need VS2013 why do I need qmake etc.. or is that VS2013 is only needed for linking ? may apologies I am a little confused.

**Comment by trashkalmar** [comment]
Honestly, I'm not sure about VS2010, but it may be OK. And yes, it is needed for linking only.

**Comment by stevegdavis** [comment]
I have just installed Cynwin and then did the gradlew build in AS as before but I still have the same error:-
 'command 'bash'' etc..
Can you look at my line added to detect_qmake.sh KNOWN_QMAKE_PATHS array:-

c:/Qt/5.5/mingw492_32/bin/qmake \

where qt is installed on drive C: and omim-master source is on drive M:
maybe that is wrong ? Btw I can run qmake from the command line so I know that it is installed.
Thanks.

**Comment by trashkalmar** [comment]
You have to specify path in Cygwin notation. ie: /cygdrive/c/Qt/5.5/...
QT should be installed with VS support but you selected mingw. It seems you have to change it to VS.

**Comment by stevegdavis** [comment]
OK sorry missed that, added qt msvc2010 bits and changed the qmake path with proper syntax, but no joy still getting the same error.

**Comment by deathbaba** [comment]
As far as I remember, there were bugs with cygwin when project was not on drive C:

**Comment by stevegdavis** [comment]
OK thanks for that I will put the source on drive c: and see what happens. Cheers.

**Comment by stevegdavis** [comment]
Does seem that there maybe an issue with cygwin when project not on drive C so I have copied everything onto C:.

I am now able to solve the "A problem occurred starting process 'command 'bash'' error, but hard as I tried I cannot get the gradle build to 'see' bash in C:\cygwin64\bin even adding it to the Windows PATH doesn't work so I changed the build.gradle file on the line that says:-
task cppLibsBuildDebug(type: Exec, description: 'Building DEBUG version of static C++ libraries') {
    commandLine 'bash', '../tools/autobuild/android.sh', 'debug'
}
to (for cppLibsBuildDebug only at the moment) :-
task cppLibsBuildDebug(type: Exec, description: 'Building DEBUG version of static C++ libraries') {
  if (System.properties['os.name'].toLowerCase().contains('windows'))
    commandLine 'cmd', '/c', 'cppLibsBuildDebug.bat'
  else
    commandLine 'bash', '../tools/autobuild/android.sh', 'debug'
}
and the cppLibsBuildDebug.bat file has in it:-
echo off
bash /cygdrive/c/SRC/MAPS.ME/omim-master/tools/autobuild/android.sh debug
I have also added to the Windows PATH variable:-
C:\cygwin64\bin;C:\Qt\5.5\msvc2010\bin

Now it gets a bit further but fails, see attached file:-
[cppBuildLibsDebug_Output.txt](https://github.com/mapsme/omim/files/31893/cppBuildLibsDebug_Output.txt)

Could someone have a look at the output file ?
Thanks

**Comment by stevegdavis** [comment]
Does seem that there maybe an issue with cygwin when project not on drive C so I have copied everything onto C:.

Now it gets a bit further but fails seems to have the same error as #428 ie. ERROR while building armeabi-v7a-hard config.
I have installed ndk r10e and I have changed local.properties to reflect that.
Attached build output file:-
[cppLibsBuildDebug-output.txt](https://github.com/mapsme/omim/files/32119/cppLibsBuildDebug-output.txt)

Could someone have a look at the output file ?
Thanks

**Comment by stevegdavis** [comment]
Managed to do a build under Windows 7 but I had to use cmake instead of make as the latest version of make installed with Cygwin doesn't like colon (eg. C: in Windows path) because the colon is used to separate the target from its dependencies and gives "Multiple target patterns error" but cmake does accept colon with no error(s).

I executed the following:-
gradlew clean cppLibsBuildRelease - that took 1hr 14mins
then
gradlew assembleWebRelease - that took 14mins, but at the end it said:-

58 errors, 0 warnings
                                                                                                                                                                                                                                                                   :
assembleWebRelease                   
               
BUILD SUCCESSFUL

58 errors but BUILD SUCCESSFUL what's that all about !, so it doesn't seem right.

I then did assembleGoogleRelease it built OK but still got the same message ie. 58 errors 0 warnings BUILD SUCCESSFUL

Haven't tried the resultant .apk files on a target yet.
